config/build.documentation.ts

```diff
@@ -161,13 +161,13 @@ function getExtractedDataFromSnippet(
             const whitespaceCountOnFirstLine = spaceFollowedByWordsRegex.exec(fullSnippetTextArray[0])[1].length;
 
             // Place snippet location as comment.
-            const editedFilename = filename.substr(filename.lastIndexOf('samples')).replace(/\\/g, '/');
-            text = '// Link to full sample: https://raw.githubusercontent.com/OfficeDev/office-js-snippets/prod/' + editedFilename + '\n';
+            const editedFilename = filename.substring(filename.lastIndexOf('samples')).replace(/\\/g, '/');
+            text = '// Link to full sample: https://raw.githubusercontent.com/OfficeDev/office-js-snippets/prod/' + editedFilename + '\n\n';
 
             text += indented
                 .map(line => {
-                    if (line.substr(0, whitespaceCountOnFirstLine).trim().length === 0) {
-                        return line.substr(whitespaceCountOnFirstLine);
+                    if (line.substring(0, whitespaceCountOnFirstLine).trim().length === 0) {
```

Comment:
`substr` and `substring` have different behaviors. The former treats the second parameter as a length, whereas the latter uses the second parameter as an end index.

Reply:
It seems like [substr is deprecated](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr). I'll update how we're using substring properly.

Reply:
Since the swap didn't lead to any spacing changes, I'm thinking to leave this as is.

Reply:
Maybe that implies this line is never being hit? Not important for this PR, but strange that there were no consequences.

Reply:
I made some changes locally and it does get hit. There were no changes because we're returning the same value as the start index in both cases. I could update the if condition to use the index instead of the length but the outcome is the same.